Add copyright part to header check

ahornace opened the pull request:
fixes #3300

Thanks.
<!--
Thank you for proposing a contribution to the {OpenGrok project. In order to accept changes from the "outside world", all contributors should "sign" the [Oracle Contributor Agreement](www.oracle.com/technetwork/community/oca-486395.html) . 
OCA basically means that you won't be sueing Oracle over code you donate to {OpenGrok project (and similar way, that Oracle won't sue you over your patch :-D ).
If you have already signed OCA or are from Oracle, then ignore this message, you just need to sign once for all patches to {OpenGrok.
Alternative is to provide a written acceptance of OCA line into the comment of specific pull request (and ideally sign, scan and mail back OCA to Oracle in parallel), but this simple acceptance is only valid for single pull request.
-->

idodeclare commented:
CDDL 3.3 "You may not remove or alter any copyright, patent or trademark notices contained within the Covered Software, or any notices of licensing or any descriptive text giving attribution to any Contributor or the Initial Developer."

tulinkry reviewed (approved):
Woah!

tulinkry commented on dev/checkstyle/fileheader.txt:
```diff
@@ -1,19 +1,23 @@
+/\*
+ \* (Portions )?Copyright \(c\) \d\d\d\d(, \d\d\d\d)?, [^0-9].*\.
+ \*/
+package .*;
```
what is the purpose of this "package.*"?

ahornace replied:
Basically it only prevents any blank lines between the end of copyright comment and the `package` definition.

vladak commented:
> CDDL 3.3 "You may not remove or alter any copyright, patent or trademark notices contained within the Covered Software, or any notices of licensing or any descriptive text giving attribution to any Contributor or the Initial Developer."

yep, this should be done incrementally for each change. Different model is needed. Github Checks ?

vladak reviewed (changes requested):
avoid changing the actual copyrights

ahornace commented:
> yep, this should be done incrementally for each change. Different model is needed. Github Checks ?

Let's be reasonable here. I'm not changing anything important (replacing `-` with `,` or adding `(c)`at most). This basically means that we have to carry the iron ball chained to our legs for the whole eternity.

The thing I can do is to add every file touched by non Oracle employees to suppressions and then every other file would force these restrictions. That means new files as well. But this is far from ideal.

tulinkry commented:
As most of the copyrights which are not connected to Oracle are from Chris, maybe we could ask him for a consent to change the format of the copyrights and put only the rest into suppressions.

vladak commented:
> > yep, this should be done incrementally for each change. Different model is needed. Github Checks ?
> 
> Let's be reasonable here. I'm not changing anything important (replacing `-` with `,` or adding `(c)`at most). This basically means that we have to carry the iron ball chained to our legs for the whole eternity.
> 
> The thing I can do is to add every file touched by non Oracle employees to suppressions and then every other file would force these restrictions. That means new files as well. But this is far from ideal.

I'd rather be cautious here. Changing the format (dashes to commas and similar) is probably fine. Adding new copyright headers I am not sure.

ahornace commented:
> Adding new copyright headers I am not sure.

Could you be please more specific? I've added a few instances where the copyright was missing and I think I removed Trond from one file since he was employee of Sun at that time and it seemed weird -> nor @tulinkry nor @tarzanek are adding their copyright to files they changed while being employees.

However, this would still be related to at most 10s of files which I'm willing to keep as they are and add them to suppressions.

vladak commented:
> > Adding new copyright headers I am not sure.
> 
> Could you be please more specific? I've added a few instances where the copyright was missing and I think I removed Trond from one file since he was employee of Sun at that time and it seemed weird -> nor @tulinkry nor @tarzanek are adding their copyright to files they changed while being employees.
> 
> However, this would still be related to at most 10s of files which I'm willing to keep as they are and add them to suppressions.

Sounds good.

vladak commented:
Thanks !
